src/mds/nameserver2/curvefs.cpp

```diff
@@ -1251,9 +1258,15 @@ StatusCode CurveFS::GetOrAllocateSegment(const std::string & filename,
             return  StatusCode::kSegmentNotAllocated;
         } else {
             // TODO(hzsunjianliang): check the user and define the logical pool
+            if (!fileInfo.has_poolsetname()) {
+                fileInfo.set_poolsetname("ssdPoolset1");
+            }
+            Poolset poolset;
+            topology_->GetPoolset(fileInfo.poolsetname(), &poolset);
             auto ifok = chunkSegAllocator_->AllocateChunkSegment(
                             fileInfo.filetype(), fileInfo.segmentsize(),
-                            fileInfo.chunksize(), offset, segment);
+                            fileInfo.chunksize(), poolset.GetType(),
```

Comment:
noop, segment should be allocated by poolset name instead of type, because we can have more than one poolsets with same type.

Reply:
> noop, segment should be allocated by poolset name instead of type, because we can have more than one poolsets with same type.

fix

Reply:
> noop, segment should be allocated by poolset name instead of type, because we can have more than one poolsets with same type.

fix, parameter has been changed to poolsetName